### DISCUSSION TITLE
签发token的Route如何选择Consumer

### DISCUSSION THREAD

**Discussion by hongbofan** [apache/apisix#8658] [Q&A] [answered]
根据教程配置一个获取token的路由
curl http://127.0.0.1:9180/apisix/admin/routes/jas \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/apisix/plugin/jwt/sign",
    "plugins": {
        "public-api": {}
    }
}'

然后创建两个不同的jwt-auth Consumer,测试后发现只有最新创建的会起作用,Router如何去指定使用某一个Consumer?

**Comment by tokers** [comment, marked as answer]
The args `key` must be specified when you call `/apisix/plugin/jwt/sign`, and this is how APISIX find the corresponding consumers (via the `key` mapping).

**Reply by hongbofan** [reply]
thx

**Reply by hongbofan** [reply]
这里就有一个疑问，为什么不通过Consumer username或者Consumer id 之类的作为关联 而是用key，这样导致不同的加密算法的key不能设置为一致了吗？

**Reply by hongbofan** [reply]
那当我获取完token之后
根据示例添加的Router
curl http://127.0.0.1:9180/apisix/admin/routes/1 \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "methods": ["GET"],
    "uri": "/index.html",
    "plugins": {
        "jwt-auth": {}
    },
    "upstream": {
        "type": "roundrobin",
        "nodes": {
            "127.0.0.1:1980": 1
        }
    }
}'

当我访问/index.html 又是如何指定某一个jwt-auth?

**Reply by tokers** [reply]
> 这里就有一个疑问，为什么不通过Consumer username或者Consumer id 之类的作为关联 而是用key，这样导致不同的加密算法的key不能设置为一致了吗？

The `key` is not an attribute of the consumer. It's part of the JWT auth credentials. We also use the credentials on the consumer group, so we won't use any consumer-related identities.

**Reply by tokers** [reply]
> 那当我获取完token之后 根据示例添加的Router curl http://127.0.0.1:9180/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d ' { "methods": ["GET"], "uri": "/index.html", "plugins": { "jwt-auth": {} }, "upstream": { "type": "roundrobin", "nodes": { "127.0.0.1:1980": 1 } } }'
> 
> 当我访问/index.html 又是如何指定某一个jwt-auth?

Can't understand your question. We never say how to specifiy a jwt-auth. A route which enables the JWT auth will pass the JWT verification if the JSON web token is valid. If you want to allow only a few consumers to access this route, pelase use consumer-restriction plugin to setup an ACL list.
